src/CommitmentsV2/SFA.DAS.CommitmentsV2/Application/Queries/GetApprenticeshipsFilterValues/GetApprenticeshipsFilterValuesQueryHandler.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using SFA.DAS.CommitmentsV2.Data;
+using SFA.DAS.CommitmentsV2.Types;
+
+namespace SFA.DAS.CommitmentsV2.Application.Queries.GetApprenticeshipsFilterValues
+{
+    public class GetApprenticeshipsFilterValuesQueryHandler : IRequestHandler<GetApprenticeshipsFilterValuesQuery, GetApprenticeshipsFilterValuesResponse>
+    {
+        private readonly IProviderCommitmentsDbContext _dbContext;
+
+        public GetApprenticeshipsFilterValuesQueryHandler(IProviderCommitmentsDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<GetApprenticeshipsFilterValuesResponse> Handle(GetApprenticeshipsFilterValuesQuery request, CancellationToken cancellationToken)
+        {
+           var dbTasks = new []{
```

Comment:
Have we tested the performance difference between executing multiple db queries vs one query whose results could satisfy all filter requirements?

Reply:
In short - not yet but will be, the idea was that this would be cached for a period of time as the values in the filter shouldn't change as often